### PR TITLE
try/except the RFI loading

### LINKED ISSUE #127: [Bug]: RFI data never loaded, always raises warning for zip files
### Checked for duplicates

Yes - I've already checked

### Describe the bug

Zipfiles which contain the RFI data don't get loaded. They always raise the warning that the files don't exist.
```
(mapping-311) [staniewi@aurora COMPASS]$ unzip -l tests/data/S1A_IW_SLC__1SDV_20221016T015043_20221016T015111_045461_056FC0_6681.zip | grep rfi
        0  02-01-2023 13:39   S1A_IW_SLC__1SDV_20221016T015043_20221016T015111_045461_056FC0_6681.SAFE/annotation/rfi/
     7301  02-01-2023 14:28   S1A_IW_SLC__1SDV_20221016T015043_20221016T015111_045461_056FC0_6681.SAFE/annotation/rfi/rfi-s1a-iw3-slc-vv-20221016t015045-20221016t015110-045461-056fc0-006.xml
     7301  02-01-2023 14:28   S1A_IW_SLC__1SDV_20221016T015043_20221016T015111_045461_056FC0_6681.SAFE/annotation/rfi/rfi-s1a-iw2-slc-vh-20221016t015044-20221016t015109-045461-056fc0-002.xml
     6958  02-01-2023 14:28   S1A_IW_SLC__1SDV_20221016T015043_20221016T015111_045461_056FC0_6681.SAFE/annotation/rfi/rfi-s1a-iw1-slc-vv-20221016t015043-20221016t015111-045461-056fc0-004.xml
     7301  02-01-2023 14:28   S1A_IW_SLC__1SDV_20221016T015043_20221016T015111_045461_056FC0_6681.SAFE/annotation/rfi/rfi-s1a-iw3-slc-vh-20221016t015045-20221016t015110-045461-056fc0-003.xml
     7301  02-01-2023 14:28   S1A_IW_SLC__1SDV_20221016T015043_20221016T015111_045461_056FC0_6681.SAFE/annotation/rfi/rfi-s1a-iw2-slc-vv-20221016t015044-20221016t015109-045461-056fc0-005.xml
     6958  02-01-2023 14:28   S1A_IW_SLC__1SDV_20221016T015043_20221016T015111_045461_056FC0_6681.SAFE/annotation/rfi/rfi-s1a-iw1-slc-vh-20221016t015043-20221016t015111-045461-056fc0-001.xml

```

```
(mapping-311) [staniewi@aurora COMPASS]$ s1_info tests/data/S1A_IW_SLC__1SDV_20221016T015043_20221016T015111_045461_056FC0_6681.zip
Found 1 Sentinel-1 SLC products.
Bursts in tests/data/S1A_IW_SLC__1SDV_20221016T015043_20221016T015111_045461_056FC0_6681.zip:
--------------------------------------------------------------------------------
/u/aurora-r0/staniewi/repos/s1-reader/src/s1reader/s1_reader.py:683: UserWarning: No RFI metadata found at S1A_IW_SLC__1SDV_20221016T015043_20221016T015111_045461_056FC0_6681.SAFE/annotation/rfi/rfi-s1a-iw1-slc-vv-20221016t015043-20221016t015111-045461-056fc0-004.xml.
  warnings.warn(f"No RFI metadata found at {rfi_annotation_path}.")
...
```

https://github.com/isce-framework/s1-reader/blob/559f026e5a47f81e010a0bc47d73dc15afc0b1bd/src/s1reader/s1_reader.py#L673-L675

These lines check if a files exists, but it's inside the zip file.

### What did you expect?

- No warning should be printed when there's not an RFI file available, since everything pre-2021 has no RFI and it's not a problem
- we should add a unit test for an example with RFI loading as a minimum smoke test

### Reproducible steps

```bash
`s1_info` on any .zip file will print a `UserWarning`
```

### Environment

_No response_

### PR DESCRIPTION
**Pull request by scottstanie**
Closes #127 

Tthe new zip file has only vv rfi/annotation for iw2, so it's ~200 kb zipped.

### CODE REVIEW

File: src/s1reader/s1_reader.py [hunk 1]
```diff
@@ -672,15 +672,14 @@ def burst_from_xml(annotation_path: str, orbit_path: str, tiff_path: str,
...
             with open_method(rfi_annotation_path, 'r') as f_rads:
                 tree_rads = ET.parse(f_rads)
                 burst_rfi_info_swath = SwathRfiInfo.from_et(tree_rads,
                                                             tree_lads,
                                                             ipf_version)
 
-        else:
-            warnings.warn(f"No RFI metadata found at {rfi_annotation_path}.")
+        except (FileNotFoundError, KeyError):
             burst_rfi_info_swath = None
```
Comment (seongsujeong) [inline src/s1reader/s1_reader.py:683]:
```suggestion
        except (FileNotFoundError, KeyError):
            if ipf_version >= RFI_INFO_AVAILABLE_FROM:
                warnings.warn(f"RFI annotation expected (IPF version={ipf_version}"
                              f" >= {RFI_INFO_AVAILABLE_FROM}), but not loaded.")
            burst_rfi_info_swath = None
```

I think it's still be helpful to display warning message to early notice the bug.
The constant `RFI_INFO_AVAILABLE_FROM` is defined in `s1_annotation.py`. You might need to import that from there.

**Review by seongsujeong — COMMENTED** [review]
Looks good to me. I have one suggestion as in the below - Seongsu

**Review by seongsujeong — APPROVED** [review]
LGTM. Thanks for the fix! @scottstanie

**Review by vbrancat — APPROVED** [review]
LGTM!
